api/proto/teleport/recordingencryption/v1/recording_encryption.proto

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.recordingencryption.v1;
+
+import "teleport/header/v1/metadata.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/recordingencryption/v1;recordingencryptionv1";
+
+// KeyState represents the possible states a WrappedKey can be in.
+enum KeyState {
+  // Default KeyState
+  KEY_STATE_UNSPECIFIED = 0;
+  // KEY_STATE_ACTIVE marks a key in good standing.
+  KEY_STATE_ACTIVE = 1;
+  // KEY_STATE_ROTATING marks a key as waiting for its owning auth server to
+  // rotate it.
+  KEY_STATE_ROTATING = 2;
+  // KEY_STATE_ROTATED marks a key as fully rotated.
+  KEY_STATE_ROTATED = 3;
+}
+
+// WrappedKey wraps the private key of a recording encryption key pair using a
+// separate asymmetric keypair.
+message WrappedKey {
+  // RecordingEncryptionPair is the asymmetric keypair used with age to encrypt
+  // and decrypt filekeys. The private key is encrypted using the KeyEncryptionPair's
+  // public key and has to be decrypted before recording decryption operations can
+  // be fulfilled.
+  types.EncryptionKeyPair recording_encryption_pair = 1;
+  // KeyEncryptionPair is the asymmetric keypair used to wrap (encrypt) the
+  // RecordingEncryptionPair's private key.
+  types.EncryptionKeyPair key_encryption_pair = 2;
+  // State represents what state a given WrappedKey is currently in. Primarily used
+  // to identify what phase of a rotaiton a given key is currently in.
+  KeyState state = 3;
```

Comment:
Using enums in proto brings some UX issues to the Teleport terraform provider.
When using those, we have to write the numerical identifier instead of `ACTIVE/ROTATING/ROTATED`.

I'm not sure this view is popular, but lately I've favored string representations instead of Enums.

Up to you (and whoever also reviews this), just trying to raise your attention to this.

Note: this is a terraform provider issue, but while we don't fix it, this will have a negative impact in UX.

Reply:
I think this field actually started as a string in the RFD but was replaced with an enum after a couple of rounds of feedback. The enum might make a bit more sense in this case since `RecordingEncryption` state is meant to be internally created and managed within the auth service. The `KeyState` values shouldn't be exposed in a way that would expect to impact the TF provider or even `tctl get` since the only planned RPCs are for powering the encryption key rotation sub command.

That said I might go ahead and just omit for the time being.